### PR TITLE
Properly handle headers with empty values

### PR DESCRIPTION
**Pull request by halter73**

### CODE REVIEW

**Comment by Tratcher** [comment]
:shipit: Have you also filed a bug against HttpClient for sending the empty header?

**Comment by halter73** [comment]
I just filed an issue about the empty header: https://github.com/dotnet/corefx/issues/4065

I might have to add some more details to that issue and file another issue about ByteArrayContent, but I'll do that tomorrow when I can hopefully get my dev environment behaving again.

**Comment by Tratcher** [comment]
I'd guess ByteArrayContent is a manifestation of the same issue because it doesn't include a content-type header by default. I'd be curious to see the wireshark trace.

**Comment by troydai** [comment]
:shipit:

**Comment by halter73** [comment]
@lodejard I know you aren't a huge fan of creating an extra copy of the scan iterator to implement this fix, but are you OK with getting this PR in as is for RC1?

**Comment by lodejard** [comment]
:shipit:

File: src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs [hunk 1]
```diff
@@ -739,7 +752,7 @@ private bool TakeMessageHeaders(SocketInput input)
                         }
 
                         consumed = scan;
-                        _requestHeaders.Append(name.Array, name.Offset, name.Count, value);
+                        requestHeaders.Append(name.Array, name.Offset, name.Count, value);
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: Why this change

Comment (halter73) [inline src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs]: For testing.
